assembly/nearEntry.ts

```diff
@@ -1,429 +0,0 @@
-  assert(val instanceof Str, "Value with Key: " + name + " is not a string or null");
-  return (<Str>val)._str;
-}
-
-function decodeArray<T>(val: Value, name: string): Array<T> {
-  assert(val instanceof Arr, "Value with Key: " + name + " is not an array or null.");
-  const res = new Array<T>();
-  const arr = (<Arr>val)._arr;
-  for (let i: i32 = 0; i < arr.length; i++) {
-    let item: T = decode<T, Value>(arr[i]);
-    res.push(item);
-  }
-  return res;
-}
-
-function isReallyNullable<T>(): bool {
-  return isReference<T>() || isArrayLike<T>() || isNullable<T>() || isString<T>();
-}
-
-//@ts-ignore
-@global
-function decode<T, V = Uint8Array>(buf: V, name: string = ""): T {
-  const buffer = <Value>(buf instanceof Uint8Array ? JSON.parse(buf) : buf);
-  var val: Value;
-  if (buffer instanceof Obj && name != "") {
-    const obj: Obj = <Obj>buffer;
-    let res = obj.get(name);
-    if (res == null) {
-      //@ts-ignore
-      return <T>null;
-    }
-    val = res;
-  }else {
-    val = <Value> buffer;
-  }
-  if (val instanceof Null) {
-    assert(isReallyNullable<T>(), "Key: " + name + " with type " + nameof<T>() + "is not nullable.");
-    //@ts-ignore
-    return <T>null;
-  }
-  if (isString<T>()) {
-    //@ts-ignore
-    return getStr(val, name);
-  }
-  if (isBoolean<T>()) {
-    assert(val instanceof Bools, "Value with Key: " +  name + " with type " + nameof<T>()  + " is not a string");
-    //@ts-ignore
-    return (<Bools>val)._bool;
-  }
-  var value: T;
-  if (isInteger<T>()) {
-    //@ts-ignore
-    if (value instanceof u64 || value instanceof i64) {
-      assert(val instanceof Str, "Value with Key: " +  name + " with type " + nameof<T>()  + " is an 64-bit integer and is expected to be encoded as a string");
-      let str = (<Str>val)._str;
-      //@ts-ignore
-      return <T>(val instanceof u64) ? U64.parseInt(str) : I64.parseInt(str);
-    }
-    assert(val instanceof Num, "Value with Key: " +  name + " with type " + nameof<T>()  + " is not an Integer");
-    //@ts-ignore
-    return <T>(<Num>val)._num;
-  }
-  assert(isReference<T>(), name + " with type " + nameof<T>() + " must be an integer, boolean, string, object, or array");
-  if (isArrayLike<T>()) {
-    //@ts-ignore
-    if (value instanceof Uint8Array ) {
-      //@ts-ignore
-      return base64.decode(getStr(val, name));
-    }
-    //@ts-ignore
-    // assert(val instanceof Arr, "Value with Key: " +  name + " with type " + nameof<T>()  + " is expected to be an array")
-    //@ts-ignore only checking the instance
-    return <T>decodeArray<valueof<T>>(val, name);
-  }
-  //@ts-ignore
-  if (value instanceof u128) {
-    assert(val instanceof Str);
-    //@ts-ignore
-    return u128.fromString(getStr(val, name));
-  }
-  assert(val instanceof Obj, "Value with Key: " +  name + " with type " + nameof<T>()  + " is not an object or null");
-  value = instantiate<T>();
-  //@ts-ignore
-  return value.decode<Obj>(<Obj>val);
-}
```

Comment:
looks like there are fewer lines of code in index.ts. Is some code getting deleted? Is that intentional?

Reply:
Yeah I moved all of the JSON Value object classes into assemblyscript-json so it can be used outside of this project.